app/controllers/setup_controller.rb

```diff
@@ -31,17 +29,19 @@ def welcome
     @services_cidr = Pillar.value(pillar: :services_cidr) || "172.24.0.0/16"
     @api_cluster_ip = Pillar.value(pillar: :api_cluster_ip) || "172.24.0.1"
     @dns_cluster_ip = Pillar.value(pillar: :dns_cluster_ip) || "172.24.0.2"
-    @suse_registry_mirror = DockerRegistry.find_or_initialize_by(mirror: SUSE_REGISTRY_URL)
+    @suse_registry_mirror = RegistryMirror.find_or_initialize_by(
```

Comment:
You are assuming the `db:seed` will happen, right? Because if the seed doesn't happen, it will initialize a registry without a name. Might break for production env.

Reply:
Maybe instead of `registry_id` you could use `name` and `SUSE` because no other registry would exist and remove the db seed.

Reply:
that's the thing, i used a seed because that's what they're there for in rails, do you think this could collide in some way? we run the seeds in `bin/init` so we can assume it ran

Reply:
I don't have a strong opinion about this. If we guarantee that the seed won't mess up with things, I guess that's fine.

Reply:
Since you've removed the seeds, you need to change this to use `name` instead of `registry_id`.

Reply:
Still missing this...

Reply:
If migration will happen, then we will always have a registry, right? Then this change is not necessary.

Reply:
No, the migration won't happen on new installations, only on upgrades.

Reply:
@vitoravelino sorry didnt see the comments....

@ereslibre not sure what you mean, all the migrations are run on new installations as well

Reply:
No, they won't, and we've played with that in the past. A new installation will basically go through https://github.com/kubic-project/velum/blob/master/bin/init#L47 (`db:schema:load`, which will assume all migrations have been ran until the latest version of the (schema definition)[https://github.com/kubic-project/velum/pull/437/files#diff-1acd2e7e27a227829d5d14a91c863bb6L14]).

Reply:
maybe i don't get the idea here ;-) but at some point we need to run the migrations, no? if we wouldn't run them we wouldn't have any table available

Reply:
`db:schema:load` will get the `db/schema.rb` and create all tables in the database, also assuming that all migrations until the version specified in `db/schema.rb` have been run. That's why on new installations migrations won't run (it's a feature, so you on new installations only load the schema from the `db/schema.rb` that is the reflection of the database status)

Reply:
I just realized that this is the mirror and not the registry. So this line is fine. Sorry for the confusion! :sweat_smile: